src/DropdownReference/components/DropdownReference.ts

```diff
@@ -0,0 +1,120 @@
+                className: this.props.className,
+                label: this.props.labelCaption,
+                orientation: this.props.labelOrientation,
+                style: this.props.styleObject,
+                weight: this.props.labelWidth
+            }, this.renderSelector())
+            : this.renderSelector();
+    }
+
+    componentDidMount() {
+        const scrollContainer = document.querySelector(".region-content .mx-scrollcontainer-wrapper");
+        if (scrollContainer) {
+            const dropdown = document.getElementsByClassName("Select-menu-outer");
+            scrollContainer.addEventListener("scroll", () => {
+                dropdown[0] ? (dropdown[0] as HTMLElement).style.visibility = "hidden" : window.logger.warn("Dropdown not available");
+                const activeElement = document.activeElement;
+                if (activeElement) {
+                    (activeElement as HTMLElement).blur();
+                }
+            });
+        }
+    }
+
+    private renderSelector() {
+        const commonProps = {
+            clearable: this.props.isClearable,
+            disabled: this.props.isReadOnly,
+            onChange: this.props.handleOnchange,
+            ...this.createSelectorProp()
+        };
+
+        if (this.props.readOnlyStyle === "control" || (this.props.readOnlyStyle === "text" && !this.props.isReadOnly)) {
+            return createElement("div", {
+                className: "widget-dropdown-reference",
+                onClick: this.setDropdownSize
+            },
+                this.props.selectType === "normal"
+                    ? createElement(Select, {
+                        options: this.props.data,
+                        ...commonProps
+                    })
+                    : createElement(Async, {
+                        searchPromptText: this.props.minimumCharacter > 0
+                            ? `Type more than ${this.props.minimumCharacter} characters to search`
+                            : "Type to search",
+                        loadOptions: this.props.asyncData,
+                        ...commonProps
+                    }),
+                createElement(Alert, { className: "widget-dropdown-reference-alert" }, this.props.alertMessage)
+            );
+        } else {
+            return createElement("p", { className: classNames("form-control-static", "read-only-text") },
+                this.props.selectedValue ? this.props.selectedValue.label : "");
+        }
+    }
+
+    private setDropdownSize = () => {
+        const dropdown = document.getElementsByClassName("Select-menu-outer");
+        if (dropdown[0] && (dropdown[0] as HTMLElement).style.visibility !== "visible") {
+            const dropdownDimensions = dropdown[0].getBoundingClientRect();
+            if (dropdown && dropdown.length && dropdownDimensions) {
+                (dropdown[0] as HTMLElement).style.width = dropdownDimensions.width - .08 + "px";
```

Comment:
Define a constant and just change the properties instead of use `(dropdown[0] as HtmlElement)` each line.

Eg.
`const dropdownElement = dropdown[0] as HTMLElement;
                dropdownElement.style.width = dropdownDimensions.width - .08 + "px";
                dropdownElement.style.left = dropdownDimensions.left + "px";
                dropdownElement.style.top = dropdownDimensions.top + "px";
                dropdownElement.style.visibility = "visible";
                dropdownElement.style.position = "fixed";`

Reply:
That's more efficient. Thanx